nominal/sdk.py

```diff
@@ -433,22 +454,29 @@ def create_dataset_from_io(
             "relative_{unit}": relative timestamps (floats or ints),
             where {unit} is one of: nanoseconds | microseconds | milliseconds | seconds | minutes | hours | days
         """
-
         # TODO(alkasm): create dataset from file/path
 
         if isinstance(dataset, TextIOBase):
             raise TypeError(f"dataset {dataset} must be open in binary mode, rather than text mode")
-        filename = f"{name}{file_extension}"
+        urlsafe_name = urlsafe_b64encode(name.encode()).decode()
+        filename = f"{urlsafe_name}{file_extension}"
         s3_path = put_multipart_upload(self._auth_header, dataset, filename, "text/csv", self._upload_client)
-        request = ingest_api.TriggerIngest(
-            labels=list(labels),
-            properties={} if properties is None else dict(properties),
+        request = ingest_api.TriggerFileIngest(
+            destination=ingest_api.IngestDestination(
+                new_dataset=ingest_api.NewDatasetIngestDestination(
+                    labels=list(labels),
+                    properties={} if properties is None else dict(properties),
+                    channel_config=None,  # TODO(alkasm): support offsets
+                    dataset_description=description,
+                    dataset_name=name,
+                )
+            ),
             source=ingest_api.IngestSource(s3=ingest_api.S3IngestSource(path=s3_path)),
-            dataset_description=description,
-            dataset_name=name,
-            timestamp_metadata=ingest_api.TimestampMetadata(
-                series_name=timestamp_column_name,
-                timestamp_type=_timestamp_type_to_conjure_ingest_api(timestamp_column_type),
+            source_metadata=ingest_api.IngestSourceMetadata(
+                timestamp_metadata=ingest_api.TimestampMetadata(
+                    series_name=timestamp_column_name,
+                    timestamp_type=_timestamp_type_to_conjure_ingest_api(timestamp_column_type),
+                ),
             ),
         )
         response = self._ingest_client.trigger_ingest(self._auth_header, request)
```

Comment:
wait shouldn't this be `trigger_file_ingest`?

Reply:
how did this work??

Reply:
oh I guess I jsut changed it. mypy caught this, along with some other issues that I just fixed.